conf.h

```diff
@@ -0,0 +1,9 @@
+#ifndef CONF_H_INCLUDED
+#define CONF_H_INCLUDED
+//if you want queue array ,replace -linked- by -array-.
+#define linked
+#ifdef array
+//if you want to change the size ,replace 100 with your size.
+#define arraysize    (100)
+#endif

```

Comment:
The problem in this implementation is what will happen if we define not `linked` and `array` regardless to the bad name 
In this case you have to define a macro for each option and another macro to hold the option as follows:
```
#define QUEUE_ARRAY   (44)
#define QUEUE_LINKED  (77)

#define QUEUE_TYPE     QUEUE_ARRAY   
```
and for more error handling we can do sth like : 
```
#if QUEUE_TYPE != QUEUE_ARRAY  && QUEUE_TYPE != QUEUE_LINKED   

#error "Error in defining QUEUE_TYPE "
#endif    
```